Ryujinx.Ava/Modules/Updater/Updater.cs

```diff
@@ -516,13 +502,12 @@ private static async void InstallUpdate(TaskDialog taskDialog, string updateFile
                             tarStream.CopyEntryContents(outStream);
                         }
 
+                        File.SetUnixFileMode(outPath, (UnixFileMode)tarEntry.TarHeader.Mode);
```

Comment:
```suggestion
                        if (OperatingSystem.IsLinux())
                        {
                            File.SetUnixFileMode(outPath, (UnixFileMode)tarEntry.TarHeader.Mode);
                        }
```

Why not add this in a condition and remove the `Debug.Assert` above ?

Reply:
`Debug.Assert()` won't be added to release binaries and this method only gets called on Linux anyway. So we could skip the check entirely and execute the call to `SetUnixFileMode` directly.

Reply:
I'd prefer to get it under a condition and remove the assert to get the code more obvious than just add code to remove warning and assume something isn't called under specifics OS